src/Node/Internal.hs

```diff
@@ -237,20 +237,26 @@ newtype ChannelOut m = ChannelOut (NT.Connection m)
 closeChannel :: ChannelOut m -> m ()
 closeChannel (ChannelOut conn) = NT.close conn
 
--- | Write some ByteStrings to an out channel. It does not close the
---   transport when finished. If you want that, use withOutChannel or
---   withInOutChannel.
-writeChannel
-    :: ( Monad m, WithLogger m, Mockable Throw m )
-    => ChannelOut m
-    -> [BS.ByteString]
+-- | Do multiple sends on a 'ChannelOut'.
+writeMany
+    :: ( Monad m, Mockable Throw m )
+    => Word32 -- ^ Split into chunks of at most this size in bytes. 0 means no split.
+    -> ChannelOut m
+    -> LBS.ByteString
     -> m ()
-writeChannel (ChannelOut conn) chunks = do
-    res <- NT.send conn chunks
-    -- TBD error handling? Throw exception or report it as a Left?
-    case res of
-      Left err -> throw err
-      Right _  -> pure ()
+writeMany mtu (ChannelOut conn) bss =
+    NT.send conn (chop bss >>= LBS.toChunks) >>= either throw pure
```

Comment:
This isn't right. The NT.send is a vectored send, meaning it assembles a single data frame from multiple buffers. We want multiple data frames (each of a limited size). So we need to do a `mapM_ NT.send` instead, not a single `NT.send`.